### ISSUE TITLE
[ARCHETYPE-349] Property not available in config when defaultValue in descriptor contains token

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-archetype#339]
**[Petr Janata](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=petrjanata)** opened **[ARCHETYPE-349](https://issues.apache.org/jira/browse/ARCHETYPE-349?redirect=false)** and commented

There is a bug with property resolving when calling archetype:generate mojo. Default property value provided in archetype descriptor hides the value passed as parameter to mojo.

archetype-metadata.xml:

```xml
<requiredProperty key="foo">
  <defaultValue>${some.name}</defaultValue>
</requiredProperty>
```

running:

```
mvn archetype:generate ... -Dfoo=bar
...
Define value for property 'foo':  ${some.name}: :
```

Generation stops and asks for value of property foo although it was passed as parameter.

I have attached example archetype and patch that solves this issue.
Please review and focus on suspicious line (112 before patching, 116 after patching).
Basically the default value from metadata should be ignored, when value is explicitly set.

Handling nested tokens is a different story...

---

**Affects:** 2.0

**Attachments:**
- [archetype-349.patch](https://issues.apache.org/jira/secure/attachment/12713881/archetype-349.patch) (_3.92 kB_)
- [test_and_patch.tgz](https://issues.apache.org/jira/secure/attachment/12713854/test_and_patch.tgz) (_1.14 kB_)
